tests/app/oidc/test_gcp_oidc.py

```diff
@@ -0,0 +1,62 @@
+import time
+
+from cachetools.func import ttl_cache
+from freezegun import freeze_time
+from mock import Mock, patch
+
+from app.oidc.gcp_oidc import TTL, OIDCCredentialsServiceGCP
+
+TEST_SDS_OAUTH2_CLIENT_ID = "TEST_SDS_OAUTH2_CLIENT_ID"
+
+
+@patch("app.oidc.gcp_oidc.Request")
+@patch("app.oidc.gcp_oidc.fetch_id_token_credentials")
+def test_oidc_credentials_service_gcp(mock_token_fetch, mock_request):
+    oidc_credentials_service = OIDCCredentialsServiceGCP()
+
+    # fetch credentials and check the gcp call is made
+    oidc_credentials_service.get_credentials(iap_client_id=TEST_SDS_OAUTH2_CLIENT_ID)
+    mock_token_fetch.assert_called_once_with(
+        audience=TEST_SDS_OAUTH2_CLIENT_ID, request=mock_request()
+    )
+    mock_token_fetch.return_value.refresh.assert_called_once_with(mock_request())
+
+
+@freeze_time("2023-01-01T12:00:00")
+@patch("app.oidc.gcp_oidc.Request", Mock)
+@patch("app.oidc.gcp_oidc.fetch_id_token_credentials")
+def test_oidc_credentials_service_gcp_ttl(mock_token_fetch):
+    """
+    by default, TTLCache uses a cached version of time.monotonic for the timer
+    which means that mocking time with freezegun doesn't work, as it won't affect the timer
+    to work around this and test the caching, we can replace the timer with a time.monotonic lambda
+    """
+    oidc_credentials_service = OIDCCredentialsServiceGCP()
+
+    # overwrite the timer
+    oidc_credentials_service.get_credentials = ttl_cache(
+        maxsize=None,
+        ttl=TTL,
+        timer=lambda: time.monotonic(),  # pylint: disable=unnecessary-lambda
```

Comment:
The linting is genuine, you don't need a lambda, just pass it a callable, i.e `time.monotonic`, so `time=time.monotonic`